core/src/main/java/org/apache/calcite/sql/fun/SqlFirstLastValueAggFunction.java

```diff
@@ -48,7 +48,7 @@ public SqlFirstLastValueAggFunction(SqlKind kind) {
         kind,
         ReturnTypes.ARG0_NULLABLE_IF_EMPTY,
         null,
-        OperandTypes.ANY,
+        OperandTypes.ANY_OPTIONAL_ANY,
```

Comment:
Curious, what does this do?

Reply:
I think it specifies that the first value can be any data type and the second value can also be of any data type, but may also be optionally omitted.